sweep/fee_bumper.go

```diff
@@ -957,6 +949,50 @@ func (t *TxPublisher) handleTxConfirmed(r *monitorRecord, requestID uint64) {
 	t.handleResult(result)
 }
 
+// handleInitialTxError takes the error from `initializeTx` and decides the
+// bump event. It will construct a BumpResult and handles it.
+func (t *TxPublisher) handleInitialTxError(requestID uint64, err error) {
+	// We now decide what type of event to send.
+	var event BumpEvent
+
+	switch {
+	// When the error is due to a dust output, we'll send a TxFailed so
+	// these inputs can be retried with a different group in the next
+	// block.
+	case errors.Is(err, ErrTxNoOutput):
+		event = TxFailed
+
+	// When the error is due to budget being used up, we'll send a TxFailed
+	// so these inputs can be retried with a different group in the next
+	// block.
+	case errors.Is(err, ErrMaxPosition):
+		event = TxFailed
+
+	// When the error is due to zero fee rate delta, we'll send a TxFailed
+	// so these inputs can be retried in the next block.
+	case errors.Is(err, ErrZeroFeeRateDelta):
+		event = TxFailed
+
+	// Otherwise this is not a fee-related error and the tx cannot be
+	// retried. In that case we will fail ALL the inputs in this tx, which
+	// means they will be removed from the sweeper and never be tried
+	// again.
+	//
+	// TODO(yy): Find out which input is causing the failure and fail that
```

Comment:
Very important todo, I would say it even makes sense to open a tracking issue wdyt ?